pkgs/by-name/dp/dprint/package.nix

```diff
@@ -50,17 +47,14 @@ rustPlatform.buildRustPackage (finalAttrs: {
     "--skip=utils::url::test::unsafe_ignore_cert"
   ];
 
-  postInstall =
-    ''
-      rm "$out/bin/test-process-plugin"
-    ''
-    + lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''
```

Comment:
My understanding is that this idiom helps prevent issues with cross-compilation.

https://github.com/NixOS/nixpkgs/blob/8d6345916ebe3c0e93b6dafa6fec7b21aa43a827/doc/hooks/installShellFiles.section.md?plain=1#L87-L110

Reply:
you're right, I assumed there was a reason for adding it. I had `boot.binfmt.emulatedSystems = [ "aarch64-linux" ];` enabled on my system which unfortunately also works in the nix sandbox (which I assumed it won't) so the cross compilation worked.